index.js

```diff
@@ -1,68 +1,47 @@
 /* jshint node: true, -W030 */
 
 /*
-	MIT License http://www.opensource.org/licenses/mit-license.php
-	Author Kiril Vatev @catdad
-*/
+    MIT License http://www.opensource.org/licenses/mit-license.php
+    Author Kiril Vatev @catdad
+ */
 
-var path = require("path");
-var loaderUtils = require("loader-utils");
-var SourceListMap = require("source-list-map").SourceListMap;
-var CleanCSS = require("clean-css");
+var path = require('path');
+var loaderUtils = require('loader-utils');
+var CleanCSS = require('clean-css');
+var RequestShortener = require('webpack/lib/RequestShortener');
+
+module.exports = function (content, map) {
+    var config = loaderUtils.getLoaderConfig(this, 'cssRawLoader');
 
-module.exports = function(content, map) {
-	var query = loaderUtils.parseQuery(this.query);
-    
     this.cacheable && this.cacheable();
-	this.value = content;
-    
-    if (typeof map !== 'string') {
-        map = JSON.stringify(map);
-    }
-    
-    var result = [];
-    
-    var minimize = (this && this.minimize) || false;
-    
-    if(minimize) {
-		var options = Object.create(query);
-		if(query.sourceMap && map) {
-			options.sourceMap = map;
-		}
-		var minimizeResult = new CleanCSS(options).minify(content);
-		map = minimizeResult.sourceMap;
-		content = minimizeResult.styles;
-		if(typeof map !== "string") {
-			map = JSON.stringify(map);
+    this.value = content;
+
+    if (config.minimize || this.minimize && config.minimize !== false) {
+        var options = Object.create(config);
+        if (config.sourceMap || this.sourceMap && config.sourceMap !== false) {
+            options.sourceMap = typeof map === 'string' ? map : (map ? JSON.stringify(map) : true);
```

Comment:
I left the `sourceRoot` property to be blank by default since it upsets how `ExtractTextPlugin` process maps. If it's set to `webpack://`, `ExtractTextPlugin` still runs the full path through webpack's own `devtoolModuleFilenameTemplate` and the full path ends up being something like `webpack:///./webpack://~/some-module/index.css`.

I suppose we could to a `typeof options.sourceMap === 'string'` check just to enforce a string, but it should be a self-explanatory option. Other loaders such as `babel-loader` and the `UglifyJsPlugin` allow setting `sourceRoot` by a query param or in options.